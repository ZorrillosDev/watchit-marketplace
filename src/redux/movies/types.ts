export interface MoviesArgs {
  limit?: number
}
