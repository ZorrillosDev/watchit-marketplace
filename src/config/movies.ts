export const FAKE_MOVIES = [
  {
    creator: {
      address: '0x0010',
      movies: 4
    },
    owner: {
      address: '0x0022',
      movies: 9
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/11/help-50817-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  },
  {
    creator: {
      address: '0xEe99CeFF640d37Edd9cac8c7cfF4Ed4cD609f435',
      movies: 10
    },
    owner: {
      address: '0x0054',
      movies: 7
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/08/safer-at-home-46847-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  },
  {
    creator: {
      address: '0x005',
      movies: 5
    },
    owner: {
      address: '0x006',
      movies: 3
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/11/cagefighter-worlds-collide-50877-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  },
  {
    creator: {
      address: '0x007',
      movies: 2
    },
    owner: {
      address: '0x008',
      movies: 12
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/10/este-nino-es-un-trasto-50575-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  }
]
