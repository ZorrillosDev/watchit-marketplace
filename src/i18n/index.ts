import i18n from "i18next";
import {initReactI18next} from "react-i18next";
import LanguageDetector from "i18next-browser-languagedetector";

// Languages
import en from './langs/en.json'
import es from './langs/es.json'

// default lang
export const defaultLang = 'en';

// Translation
const resources = {
    en: {translation: en},
    es: {translation: es}
}


i18n
    .use(LanguageDetector)
    .use(initReactI18next)
    .init({
        resources,
        fallbackLng: defaultLang,
        interpolation: {
            escapeValue: false
        }
    })

// Humanize displaying
export const lang = [
    {value: "en", name: "English"},
    {value: "es", name: "Spanish"}

]

export default i18n;
