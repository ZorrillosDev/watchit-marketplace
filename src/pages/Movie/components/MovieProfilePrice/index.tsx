export { MovieProfilePrice as default } from '@pages/Movie/components/MovieProfilePrice/MovieProfilePriceContainer'
