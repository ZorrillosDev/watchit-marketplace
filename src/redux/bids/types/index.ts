export * from './bids';
export * from './bids.actions';
export * from './bids.state';
