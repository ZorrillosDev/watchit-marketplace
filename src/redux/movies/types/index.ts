export * from './movies'
export * from './movies.actions'
export * from './movies.state'