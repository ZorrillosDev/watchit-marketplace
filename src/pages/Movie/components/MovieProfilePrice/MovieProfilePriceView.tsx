// REACT IMPORTS
import React, { FC } from 'react'

// MUI IMPORTS
import {
  Box, BoxProps,
  Divider, Grid, GridProps, Link, styled, Typography
} from '@mui/material'

// PROJECT IMPORTS
import MovieProfileUser from '@pages/Movie/components/MovieProfileUser'
import ModalBid from '@components/Bid'
import AcceptOffer from '@pages/Movie/components/MovieProfileAcceptOffer'
import PayOffer from '@pages/Movie/components/MovieProfilePay'
import { IconEth } from '@components/Icons'
import { Translation } from '@src/i18n'
import { SxProps, Theme } from '@mui/system'
import { useNFTHolderOf, useNFTIsApprovedFor } from '@hooks/useNFTContract'
import { Movie, MovieBid } from '@state/movies/types'
import { BLACK_HOLE } from '@w3/CONSTANTS'
import { useEthers } from '@usedapp/core'

// ===========================|| MOVIE - PROFILE - PRICE - VIEW ||=========================== //

export type MovieProfilePriceViewProps = Movie & MovieBid

export const MovieProfilePriceView: FC<MovieProfilePriceViewProps> = (props): JSX.Element => {
  const { account } = useEthers()
  const holder = useNFTHolderOf(props.token)
  const approvedBid = useNFTIsApprovedFor(props.account, props.token)

  const currentHolder = holder !== undefined && holder !== BLACK_HOLE ? holder : props.creator
  const iamCurrentHolder = account?.toLowerCase() === currentHolder.toLowerCase()
  // Has bid available and is not approved?
  const availableToAcceptOffer = props.price > 0 && props.account !== undefined && !approvedBid
  const iamCurrentApprovedBidder = props.account !== undefined && account?.toLowerCase() == props.account.toLowerCase()

  return (
    <Grid item xs={12}>
      <MovieProfilePriceSectionWrapper>
        <Grid container width='50%' alignItems='center' justifyContent='center'>
          <Grid item xs={8} md={6}>
            <Grid container spacing={1}>
              <Grid item xs={12}>
                <Typography variant='body1' color='primary' fontWeight={500} textAlign='left'>
                  <Translation target='MOVIE_PROFILE_PRICE_HIGHER' />
                </Typography>
              </Grid>
              <MovieProfilePriceSection item xs={12} display='flex' alignItems='center'>
                <IconEth />
                <Typography variant='h2' color='text.primary' fontWeight={500}>
                  {props.price} ETH
                </Typography>
              </MovieProfilePriceSection>
              {/* <Grid item xs={12}> */}
              {/*  <Typography variant='body2' color='text.secondary' textAlign='left'> */}
              {/*    $ {props.fiatPrice} */}
              {/*  </Typography> */}
              {/* </Grid> */}
            </Grid>
          </Grid>
        </Grid>
        <Divider orientation='vertical' sx={{ position: 'absolute', height: '50%' }} />
        <Grid container width='50%' justifyContent='center' alignItems='center'>
          <Grid item xs={8} md={6}>
            <Grid container spacing={1}>
              <Grid item xs={12}>
                <Typography variant='body1' fontWeight={500} color='primary' textAlign='left'>
                  <Translation target='MOVIE_PROFILE_OWNER' />
                </Typography>
              </Grid>
              <Grid item xs={12} display='flex'>
                <Link href={`https://rinkeby.etherscan.io/address/${currentHolder}`} target='__blank'>
                  <MovieProfileUser address={currentHolder} />
                </Link>
              </Grid>
            </Grid>
          </Grid>
        </Grid>

        {
                    account !== undefined && !iamCurrentHolder && !approvedBid
                      ? <ModalBid buttonSx={MovieProfileOfferButtonSx} />
                      : account !== undefined && iamCurrentHolder && availableToAcceptOffer
                        ? <AcceptOffer buttonSx={MovieProfileOfferButtonSx} {...props} />
                        : account !== undefined && approvedBid && iamCurrentApprovedBidder
                          ? <PayOffer buttonSx={MovieProfileOfferButtonSx} {...props} />
                          : <></>
                }
      </MovieProfilePriceSectionWrapper>
    </Grid>
  )
}

export const MovieProfilePriceSectionWrapper = styled(Box)<BoxProps>(({ theme }) => ({
  width: '100%',
  border: '1px solid',
  borderColor: theme.palette.divider,
  borderRadius: theme.shape.borderRadius,
  display: 'flex',
  paddingTop: '1.5rem',
  paddingBottom: '3rem',
  alignItems: 'flex-start',
  justifyContent: 'center',
  position: 'relative'
}))

export const MovieProfilePriceSection = styled(Grid)<GridProps>(({ theme }) => ({
  'svg, svg *': {
    fill: theme.palette.text.primary,
    marginRight: '0.5rem',
    marginLeft: '-0.5rem'
  }
}))

const MovieProfileOfferButtonSx: SxProps<Theme> = {
  position: 'absolute',
  bottom: '-20px',
  width: '15rem',
  height: '3rem',
  borderRadius: '12px !important',
  boxShadow: '0 5px 9px rgb(0 0 0 / 20%)',
  fontWeight: 600
}
