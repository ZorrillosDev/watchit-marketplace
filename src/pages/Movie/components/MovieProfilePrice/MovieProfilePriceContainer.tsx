// REACT IMPORTS
import React, { FC } from 'react'

// PROJECT IMPORTS
import { MovieProfilePriceView } from '@pages/Movie/components/MovieProfilePrice/MovieProfilePriceView'

/* eslint-disable  @typescript-eslint/consistent-type-assertions */

// ===========================|| MOVIE - PROFILE - PRICE - CONTAINER ||=========================== //

export const MovieProfilePriceContainer: FC = (): JSX.Element => {
  return <MovieProfilePriceView price={0} owner='0x00' fiatPrice={0} />
}
