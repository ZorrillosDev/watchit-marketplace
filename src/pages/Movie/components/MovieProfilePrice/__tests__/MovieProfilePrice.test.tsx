import React from 'react'
import {mount, shallow} from 'enzyme'
import MovieProfilePrice from '@pages/Movie/components/MovieProfilePrice'
import i18n from '@src/i18n'
import {Typography} from '@mui/material'
import {Movie} from '@state/movies/types'
import {Provider} from 'react-redux'
import {store} from '@state/store'
import * as web3Core from '@usedapp/core'
/* eslint-disable  @typescript-eslint/consistent-type-assertions */

jest.mock('react-router', () => ({
    useParams: () => {
        return {id: '123'}
    }
}))

jest.mock('@usedapp/core')
jest.mock('@hooks/useNFTContract', () => ({
    useNFTHolderOf: () => {
        return '0x0'
    }
}))

describe('<MovieProfilePrice />', () => {
    beforeEach(() => {
        jest.spyOn(web3Core, 'useEthers')
            .mockImplementation(() => ({
                    active: false,
                    connector: undefined,
                    error: undefined,
                    library: undefined,
                    chainId: undefined,
                    account: undefined,
                    setError: jest.fn(),
                    deactivate: jest.fn(),
                    activateBrowserWallet: jest.fn(),
                    activate: jest.fn()
                })
            )
    })

    const movie = {
        creator: '0xEe99CeFF640d37Edd9cac8c7cfF4Ed4cD609f435',
        token: '651268735865169554630764440897259763067310892845464250243992889613221274545'
    } as Movie

    it('should render', () => {
        const component = shallow(
            <Provider store={store}>
                <MovieProfilePrice {...movie} />
            </Provider>
        )
        expect(component).toMatchSnapshot()
    })

    it('should have movie profile price valid higher price text', () => {
        const translate = i18n.t('MOVIE_PROFILE_PRICE_HIGHER')
        const component = mount(
            <Provider store={store}>
                <MovieProfilePrice {...movie} />
            </Provider>
        )
        const typography = component.find(Typography).at(0)

        expect(typography.text()).toContain(translate)
    })

    it('should have movie profile price valid price text', () => {
        const price = 0
        const component = mount(
            <Provider store={store}>
                <MovieProfilePrice {...movie} price={price}/>
            </Provider>
        )
        const typography = component.find(Typography).at(1)

        expect(typography.text()).toContain(`${price} ETH`)
    })

    it('should have movie profile price valid owner text', () => {
        const translate = i18n.t('MOVIE_PROFILE_OWNER')
        const component = mount(
            <Provider store={store}>
                <MovieProfilePrice {...movie} />
            </Provider>
        )
        const typography = component.find(Typography).at(2)

        expect(typography.text()).toContain(translate)
    })
})
