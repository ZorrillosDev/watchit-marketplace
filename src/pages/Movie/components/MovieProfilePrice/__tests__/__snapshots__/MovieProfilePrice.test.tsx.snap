// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieProfilePrice /> should render 1`] = `ShallowWrapper {}`;
